core/extension/gdextension_interface.cpp

```diff
@@ -858,23 +858,23 @@ static GDExtensionPtrUtilityFunction gdextension_variant_get_ptr_utility_functio
 //string helpers
 
 static void gdextension_string_new_with_latin1_chars(GDExtensionUninitializedStringPtr r_dest, const char *p_contents) {
-	memnew_placement(r_dest, String(p_contents));
+	String *dest = memnew_placement(r_dest, String);
+	dest->parse_latin1(StrRange<char>::from_c_str(p_contents));
 }
 
 static void gdextension_string_new_with_utf8_chars(GDExtensionUninitializedStringPtr r_dest, const char *p_contents) {
-	memnew_placement(r_dest, String);
-	String *dest = reinterpret_cast<String *>(r_dest);
+	String *dest = memnew_placement(r_dest, String);
 	dest->parse_utf8(p_contents);
 }
 
 static void gdextension_string_new_with_utf16_chars(GDExtensionUninitializedStringPtr r_dest, const char16_t *p_contents) {
-	memnew_placement(r_dest, String);
-	String *dest = reinterpret_cast<String *>(r_dest);
+	String *dest = memnew_placement(r_dest, String);
 	dest->parse_utf16(p_contents);
 }
 
 static void gdextension_string_new_with_utf32_chars(GDExtensionUninitializedStringPtr r_dest, const char32_t *p_contents) {
-	memnew_placement(r_dest, String((const char32_t *)p_contents));
+	String *dest = memnew_placement(r_dest, String);
+	dest->parse_utf32(StrRange<char32_t>::from_c_str(p_contents));
```

Comment:
For consistency, the same changes can be done in `gdextension_string_new_with_wide_chars`, `gdextension_string_new_with_utf32_chars_and_len` and `gdextension_string_new_with_wide_chars_and_len`.

Reply:
Good idea!